### PR TITLE
fix: stop only those port that are started

### PR DESCRIPTION
**Pull request by zetxx**

### CODE REVIEW

**Comment by kalinkrustev** [comment]
Then let's improve even more - stop them in opposite order.
Better push the started ports, not their indexes to portsStarted and then do when.reduce(portsStarted.reverse(), ...)

**Comment by zetxx** [comment]
should look better now

File: index.js [hunk 1]
```diff
@@ -69,14 +70,15 @@ module.exports = {
                 return all;
             }.bind(this), [])
         ).then(function(contexts) {
-            return when.reduce(contexts, function(prev, context) {
+            return when.reduce(contexts, function(prev, context, idx) {
+                portsStarted.push(context); // collect ports that are started
```
Comment (stambata) [inline index.js]: There is still no guarantee that the port had been started. This part of the code could be just:
return context.start().then(() => portsStarted.push(context));

Comment (zetxx) [inline index.js]: we don't need to be sure that the port is started, we need only port object so we can call stop on it

Comment (stambata) [inline index.js]: ok

**Pull request merged**